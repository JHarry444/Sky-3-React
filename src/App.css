.trainer {
  border: 3px, solid, black;
  margin-right: 10px;
  cursor: pointer;
  max-width: 20%;
}