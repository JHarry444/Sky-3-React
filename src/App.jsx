import './App.css';
import Footer from './components/Footer';
import Header from './components/Header';
import LoggedIn from './components/LoggedIn';
import Trainer from './components/Trainer';
import trainers from "./data/trainers.json";
import { v4 as uuidv4 } from 'uuid';

function App() {
  return (
    <div className="App">
      <Header />
      <LoggedIn loggedIn />
      <LoggedIn />

      {
        trainers.map(({ name, age, specialism, location }) => (<Trainer
          key={uuidv4()}
          name={name}
          age={age}
          specialism={specialism}
          location={location} />))
      }

      <Footer />

    </div>
  );
}

export default App;
