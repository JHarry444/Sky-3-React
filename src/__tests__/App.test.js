import App from '../App';
import { create } from 'react-test-renderer';

// path to the file - 2nd parameter is a function that RETURNS a function
jest.mock("../components/Footer", () => {
  return () => (
    <footer>
      11:00:00
    </footer>
  )
})

test("maffs", () => {
  expect(2).toEqual(1 + 1);
});


test("app matches snapshot", () => {
  const app = create(<App />);

  expect(app.toJSON()).toMatchSnapshot();

});