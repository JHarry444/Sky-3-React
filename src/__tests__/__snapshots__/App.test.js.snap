// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      This is a header
    </h1>
  </header>
  <button>
    Log Out
  </button>
  <button>
    Log In
  </button>
  <p>
    My name is 
    Jordan
  </p>
  <p>
    I am 
    30
     years old
  </p>
  <p>
    I am a 
    Software
     trainer
  </p>
  <p>
    And I live in 
    Staffordshire
  </p>
  <p>
    My name is 
    Piers
  </p>
  <p>
    I am 
    26
     years old
  </p>
  <p>
    I am a 
    DevOps
     trainer
  </p>
  <p>
    My name is 
    JB
  </p>
  <p>
    I am 
    26
     years old
  </p>
  <p>
    I am a 
    Software
     trainer
  </p>
  <footer>
    11:00:00
  </footer>
</div>
`;
